cmd/relay/relay/account.go

```diff
@@ -182,13 +193,51 @@ func (r *Relay) EnsureAccountHost(ctx context.Context, acc *models.Account, host
 	return nil
 }
 
+// Checks that account is active. If it is not active upstream, double-checks with the host and updates status if needed; otherwise returns an error.
+func (r *Relay) EnsureAccountActive(ctx context.Context, acc *models.Account) error {
+
+	// short-circuit in common case
+	if acc.IsActive() {
+		return nil
+	}
+	// NOTE: this is checking local takedown
+	if acc.Status != models.AccountStatusActive {
```

Comment:
I believe this is already checked in `acc.IsActive()`

Reply:
`acc.IsActive()` checks both "local" status (`acc.Status`: eg, takedown on the relay itself) and "upstream" status on the PDS (`acc.UpstreamStatus`).

Here we are untangling upstream vs local status.